/*
 * Copyright (c) 2016 Cisco and/or its affiliates, and 
 *                    Cable Television Laboratories, Inc. ("CableLabs")
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

message t_DhcpData {
	repeated string CCAPCores = 1;
	repeated string TimeServers = 2;
	repeated string LogServers = 3;
	required int32  TimeOffset = 4 [default = 0];
}

message t_DhcpMessage {
	enum t_Status {
		INITIATED = 0;
		UPDATED = 1;
		FAILED = 2;
	}
	enum t_Client {
		DHCPV4 = 0;
		DHCPV6 = 1;
	}
	required t_Status Status = 1;
	optional t_DhcpData DHCPData = 2;
	optional string ErrorMessage = 3;
	required string InterfaceName = 4;
	required t_Client Client = 5;
}
